staging/src/k8s.io/client-go/discovery/cached/memcache.go

```diff
@@ -96,18 +98,22 @@ func (d *memCacheClient) RESTClient() restclient.Interface {
 	return d.delegate.RESTClient()
 }
 
-// TODO: Should this also be cached? The results seem more likely to be
```

Comment:
Do you understand this TODO? What's the original concern?

This PR looks ok, but I'm not surew what this TODO tries to convey.

Reply:
This is my understanding. The TODO is considering the possible inconsistency between the resources and preferred resources, one from the cache, the other from the delegate. To cache them all could ensure the consistency. 
@caesarxuchao

Reply:
Also if we cache only half of the data and the other half (preferred resources) retrieved in real-time, we could get inconsistencies.

Reply:
Make sense.